sw/device/tests/alert_handler_reverse_ping_in_deep_sleep_test.c

```diff
@@ -0,0 +1,251 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/math.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_alert_handler.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "alert_handler_regs.h"  // Generated.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+enum {
+  // This time needs to be greater than 0.175s. See test plan for more details.
+  kTestParamWakeupThresholdUsec = 200000,
+
+  // This time is needed to cycle through all the alert pings. See test plan
+  // for more details.
+  kTestParamCycleThroughAllPingsUsec = kTestParamWakeupThresholdUsec >> 2,
+  kTestParamAlertHandlerIrqDeadlineUsec = 100,
+  kTestParamAlertHandlerPhase0EscalationDurationUsec = 100,
+  kTestParamAlertHandlerPingTimeoutUsec = 20,
```

Comment:
I think 20 us is quite a long time, I would suggest something shorter to expose the error faster.
In the register description it says:
It is recommended to set this value to the equivalent of 256 cycles of the slowest alert sender clock domain in the system (or greater).
I think the value 256 * AON clock period should be fine :)

Reply:
Actually I did the conversion wrong, 20us is fine :) Sorry for the noise.